### ISSUE TITLE
Cursor disappears when exception is raised using autospin

### ISSUE THREAD

**Issue by ragurney** [piotrmurach/tty-spinner#34]
### Describe the problem
When using `auto_spin` with `hide_cursor: true`, if an exception is raised the cursor will remain hidden on the terminal until a successful `auto_spin` is called or the user manually restores the cursor. This is kind of inconvenient during development, and would be confusing to the user if they encounter an unhandled exception.

### Steps to reproduce the problem
```ruby
#!/usr/bin/env ruby

require 'tty-spinner'

s = TTY::Spinner.new(hide_cursor: true)

s.auto_spin
sleep(2)
raise "Oh no!"
s.stop
```

### Actual behaviour
Cursor remained hidden

### Expected behaviour
A rescue in `auto_spin` restores the cursor

### Describe your environment

* OS version: macOS 10.12.6
* Ruby version: 2.5.1
* TTY::Spinner version: 0.8.0

Happy to submit a PR if this is a change you'd accept!

**Comment by piotrmurach** [comment]
Hi Ryan,

Thanks for using this gem!

Your code example demonstrates an error outside of spinner instance. I don't believe that there is any code that can be added to `TTY::Spinner` to recover from such situation and gracefully restore cursor. The error is raised outside of the `spinner` client and though it crashes the spinner it hasn't been caused by it hence even a rescue clause anywhere inside `TTY::Spinner` wouldn't recover in this situtation. Wouldn't you agree?

I've added couple commits that rescue and ensure that cursor is restored when failure happens inside of `TTY::Spinner`. So one common situation that will be gracefully handled from now on is if you do:

```ruby
require 'tty-spinner'

s = TTY::Spinner.new(hide_cursor: true)

s.auto_spin do
  sleep(2)
  raise "Oh no!"
end
s.stop
```

Otherwise your best bet when using hidden cursor is to surround the client itself with rescue clause:

```ruby
require 'tty-spinner'

def spinning_animation
  s = TTY::Spinner.new(hide_cursor: true)
  s.auto_spin
  sleep(2)
  raise "Oh no!"
  s.stop
rescue
  print TTY::Cursor.show
end
```

And this is the reason why I've decided to make it an option and not the default as I belive that 'out-of-the-box' behaviour is less confusing and more bulletproof. Any thoughts?
